### PR TITLE
Backport #24020 to openssl-3.1

### PR DESCRIPTION
**Pull request by nhorman**
Backport #24020 to 3.1

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: test/prov_config_test.c [hunk 1]
```diff
@@ -70,6 +72,51 @@ static int test_recursive_config(void)
...
+{
+    OSSL_LIB_CTX *ctx = NULL;
+    OSSL_PROVIDER *prov;
+    int testresult = 0;
+    struct stat sbuf;
+    char *module_path = getenv("OPENSSL_MODULES");
+    char *full_path = NULL;
+    int rc;
+
+    full_path = OPENSSL_zalloc(strlen(module_path) + strlen(P_TEST_PATH) + 1);
```
Comment (tmshort) [inline test/prov_config_test.c:86]: Coverity is complaining that `module_path` could be NULL. I know this is a backport; and that it should be fixed in the master branch first.

**Comment by nhorman** [comment]
coverity fix available in https://github.com/openssl/openssl/pull/24208

File: test/prov_config_test.c (hunk 1 above)
Comment (nhorman) [reply test/prov_config_test.c:89]: This has been fixed in master via https://github.com/openssl/openssl/pull/24208 , and has been cherry-picked to this branch

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]
Technically speaking 3.1 is now in security fix only mode...

**Comment by t8m** [comment]
Merged to the 3.1 and 3.0 branches. Thank you.

**Pull request closed without merging**